isteven-multi-select.js

```diff
@@ -1,1111 +1,1377 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
  *
- * The above copyright notice and this permission notice shall be included in all 
+ * The above copyright notice and this permission notice shall be included in all
  * copies or substantial portions of the Software.
- * 
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE 
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  * --------------------------------------------------------------------------------
  */
 
-'use strict'
-
-angular.module( 'isteven-multi-select', ['ng'] ).directive( 'istevenMultiSelect' , [ '$sce', '$timeout', '$templateCache', function ( $sce, $timeout, $templateCache ) {
-    return {
-        restrict: 
-            'AE',
-
-        scope: 
-        {   
-            // models
-            inputModel      : '=',
-            outputModel     : '=',
-
-            // settings based on attribute
-            isDisabled      : '=',
-
-            // callbacks
-            onClear         : '&',  
-            onClose         : '&',
-            onSearchChange  : '&',  
-            onItemClick     : '&',            
-            onOpen          : '&', 
-            onReset         : '&',  
-            onSelectAll     : '&',  
-            onSelectNone    : '&',  
-
-            // i18n
-            translation     : '='   
+('use strict');
+
+angular
+  .module('isteven-multi-select', ['ng', 'vs-repeat'])
+  .directive('istevenMultiSelect', [
+    '$sce',
+    '$timeout',
+    '$templateCache',
+    function($sce, $timeout, $templateCache) {
+      return {
+        restrict: 'AE',
+
+        scope: {
+          // models
+          inputModel: '=',
+          outputModel: '=',
+
+          // settings based on attribute
+          isDisabled: '=',
+
+          // callbacks
+          onClear: '&',
+          onClose: '&',
+          onSearchChange: '&',
+          onItemClick: '&',
+          onOpen: '&',
+          onReset: '&',
+          onSelectAll: '&',
+          onSelectNone: '&',
+          onScrollBottom: '&',
```

Comment:
`onScrollBottom`, `onBeginFiltering`, `onEndFiltering` are all new

Reply:
Are we still using these now?